samcli/lib/utils/hash.py

```diff
@@ -41,22 +41,28 @@ def file_checksum(file_name: str, hash_generator: Any = None) -> str:
         return cast(str, hash_generator.hexdigest())
 
 
-def dir_checksum(directory: str, followlinks: bool = True, ignore_list: Optional[List[str]] = None) -> str:
+def dir_checksum(
+    directory: str, followlinks: bool = True, ignore_list: Optional[List[str]] = None, hashing_method: str = "md5"
+) -> str:
     """
 
     Parameters
     ----------
     directory : A directory with an absolute path
     followlinks: Follow symbolic links through the given directory
     ignore_list: The list of file/directory names to ignore in checksum
+    hashing_method: The hashing method used to generate checksum
 
     Returns
     -------
-    md5 checksum of the directory.
+    checksum hash of the directory.
 
     """
     ignore_set = set(ignore_list or [])
-    md5_dir = hashlib.md5()
+    if hashing_method == "sha256":
+        hash_dir = hashlib.sha256()
+    else:
+        hash_dir = hashlib.md5()
```

Comment:
Can we use the same way which is used in `file_checksum` method (check `hash_generator` parameter)? Instead of passing the string of the method, we can pass the method itself.